docs/topics/power-assert.md

```diff
@@ -0,0 +1,380 @@
+    // List of source sets for the Power-assert plugin
+    includedSourceSets = ["commonMain", "jvmMain", "jsMain", "nativeMain"]
+}
+```
+
+</tab>
+</tabs>
+
+## Use the plugin
+
+This section contains examples of using the Power-assert compiler plugin.
+
+See the complete code of the build script file `build.gradle.kts` for all these examples:
+
+```kotlin
+import org.jetbrains.kotlin.gradle.ExperimentalKotlinGradlePluginApi
+
+plugins {
+    kotlin("jvm") version "%kotlinVersion%"
+    kotlin("plugin.power-assert") version "%kotlinVersion%"
+}
+
+group = "org.example"
+version = "1.0-SNAPSHOT"
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation(kotlin("test"))
+}
+
+tasks.test {
+    useJUnitPlatform()
+}
+
+@OptIn(ExperimentalKotlinGradlePluginApi::class)
+powerAssert {
+    functions = listOf("kotlin.assert", "kotlin.test.assertEquals", "kotlin.test.assertTrue", "kotlin.test.assertNull", "kotlin.require")
+}
+```
+{initial-collapse-state="collapsed"}
+
+### Assert function
+
+Consider the following test with the `assert()` function:
+
+```kotlin
+import kotlin.test.Test
+
+class SampleTest {
+
+    @Test
+    fun testFunction() {
+        val hello = "Hello"
+        val world = "world!"
+        assert(hello.length == world.substring(1, 4).length) { "Incorrect length" }
+    }
+}
+```
+
+If you run the `testFunction()` test with the Power-assert plugin enabled, you get the explicit failure message:
+
+```text
+Incorrect length
+assert(hello.length == world.substring(1, 4).length) { "Incorrect length" }
+       |     |      |  |     |               |
+       |     |      |  |     |               3
+       |     |      |  |     orl
+       |     |      |  world!
+       |     |      false
+       |     5
+       Hello
+```
+
+To get a more complete error message, always inline the variable into the test function parameters.
+Consider the following test function:
+
+```kotlin
+class ComplexExampleTest {
+
+   data class Person(val name: String, val age: Int)
+
+   @Test
+   fun testComplexAssertion() {
+       val person = Person("Alice", 30)
+       val isValidName = person.name.startsWith("A") && person.name.length > 3
+       val isValidAge = person.age > 20 && person.age < 29
+       assert(isValidName && isValidAge)
+   }
+}
+```
+
+The output of the executed code doesn't provide enough information to find the cause of the problem:
+
+```text
+Assertion failed
+assert(isValidName && isValidAge)
+       |              |
+       |              false
+       true
+```
+
+Inline the variable into the `assert()` function:
+
+```kotlin
+class ComplexExampleTest {
+
+    data class Person(val name: String, val age: Int)
+
+    @Test
+    fun testComplexAssertion() {
+        val person = Person("Alice", 30)
+        assert(person.name.startsWith("A") && person.name.length > 3 && person.age > 20 && person.age < 29)
```

Comment:
Consider using `Person("Alice", 10)` or some age less or equal to 20. This will result in the following output (manually typed...), which shows that power-assert handles boolean short-circuiting correctly.

```text
Assertion failed
assert(person.name.startsWith("A") && person.name.length > 3 && person.age > 20 && person.age < 29)
       |      |    |                  |      |    |      |      |      |   |
       |      |    |                  |      |    |      |      |      |   false
       |      |    |                  |      |    |      |      |      10
       |      |    |                  |      |    |      |      Person(name=Alice, age=10)
       |      |    |                  |      |    |      true
       |      |    |                  |      |    5
       |      |    |                  |      Alice
       |      |    |                  Person(name=Alice, age=10)
       |      |    true
       |      Alice
       Person(name=Alice, age=10)
```

You could then maybe add a note about using `and` and `or` expressions if short-circuiting is not desired.